File: Java/DesignPatterns/src/test/java/designPatterns/structural/adapter/MainAdapterTest.java
package designPatterns.structural.adapter;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

public class MainAdapterTest {

  private Adaptee adaptee;
  private Target target;

  @BeforeEach
  void setUp() {
    adaptee = new Adaptee();
    target = new Adapter(adaptee);
  }

  @Test
  void equalsTest() {
    assertThat(target.request()).isEqualTo("Called specificRequest");
  }
}

File: Kotlin/DesignPatterns/src/test/kotlin/structural/adapter/MainAdapterTest.kt
package structural.adapter

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test

class MainAdapterTest {

  private lateinit var adaptee: Adaptee
  private lateinit var target: Target

  @BeforeEach
  internal fun setUp() {
    adaptee = Adaptee()
    target = Adapter(adaptee)
  }

  @Test
  internal fun equalsTest() {
    assertThat(target.request()).isEqualTo("Called specificRequest")
  }
}